internal/controllers/gateway/gateway_controller.go

```diff
@@ -95,7 +95,8 @@ func (r *GatewayReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	} else {
 		fromSelector := gatewayv1alpha2.NamespacesFromSelector
 		namespaces.From = &fromSelector
-		// TODO: this will be fully implemented when HTTPRoute support is added
+		// TODO: this will be fully implemented when namespacing support is complete
+		//       See: https://github.com/Kong/kubernetes-ingress-controller/issues/2080
```

Comment:
The difference between "full implementation" and "current implementation" is not visible to me after a glance at this code, nor at #2080.

Suggestion: more specific ACs on #2080 and changing the language of this TODO from "this will be fully implemented later" to "this currently does not do X but it should because Y, #2080 tracks this feature"